userspace/libscap/scap_savefile.c

```diff
@@ -259,6 +276,45 @@ static int32_t scap_write_fdlist(scap_t *handle, scap_dumper_t *d)
 	return SCAP_SUCCESS;
 }
 
+//
+// Since the process list isn't thread-safe, we at least reduce the
+// time window and write everything at once with a secondary dumper.
+// By doing so, the likelihood of having a wrong total length is lower.
+//
+scap_dumper_t *scap_write_proclist_begin(scap_t *handle)
+{
+	return scap_managedbuf_dump_create(handle);
+}
+int scap_write_proclist_end(scap_t *handle, scap_dumper_t *d, scap_dumper_t *proclist_dumper, uint32_t totlen)
+{
+	int res = SCAP_SUCCESS;
+

```

Comment:
I think we should return `SCAP_FAILURE` if the dumper is not of type `DT_MANAGED_BUF`, otherwise we could mistakely segfault I think.

Since we have the `scap_t * h` handle, i would also consider filling the `last_error` with something meaningful (not strong on this though).

Reply:
We don't really have such error checking in the other functions, so I didn't bother much...
I'll just put some asserts, if you don't mind :)